src/odemis/gui/conf/util.py

```diff
@@ -490,7 +490,10 @@ def format_choices(choices):
             logging.exception("Formatting error for %s", choices)
             choices_formatted = [(c, choice_to_str(c)) for c in choices]
     else:
-        choices_formatted = [(c, choice_to_str(c)) for c in choices]
+        if all(isinstance(c, str) for c in choices):
+            choices_formatted = [(c, c) for c in choices]
+        else:
+            choices_formatted = [(c, choice_to_str(c)) for c in choices]
```

Comment:
I'd suggest to just fix `choice_to_str()`. The current problem is that string are iterables, so it tries to put a `x` between each letters :-/
You should be able to just write something like:
```
if isinstance(choice, basestring) or not isinstance(choice, collections.Iterable):
    return unicode(choice)
else:
    return u" x ".join(unicode(c) for c in choice)
```